/************************************************************************/
/*
Copyright 2017 wtulip Inc.
License GPL
Author:crazycooler
Description:用户登录对话框
*/
/************************************************************************/
#pragma once
#include "MCommon.h"
#include <QDialog>
#include "ui_MSignIn.h"
#include "MHttp.h"

class MSignIn : public QDialog
{
	Q_OBJECT

public:
	MSignIn(QWidget *parent = Q_NULLPTR);
	~MSignIn();

public:
	/************************************************************************/
	/* 
	服务器对登录请求的应答回调
	*/
	/************************************************************************/
	void onLoginCallback(int status, const MResponseData &data);

	/************************************************************************/
	/* 
	获得用户名
	*/
	/************************************************************************/
	QString getUserName() { return m_userName; }

private:
	/************************************************************************/
	/* 
	初始化
	*/
	/************************************************************************/
	void init();

	/************************************************************************/
	/*
	处理windows的窗口消息，来实现无边框窗口的拖动功能
	*/
	/************************************************************************/
	virtual bool nativeEvent(const QByteArray &eventType, void *message, long *result);
private slots:
	/************************************************************************/
	/* 
	事件函数
	用户点击登录按钮时调用
	*/
	/************************************************************************/
	void signIn();

	/************************************************************************/
	/*
	事件函数
	用户点击注册按钮时调用
	*/
	/************************************************************************/
	void signUp();
private:
	Ui::MSignIn ui;
	//用户名
	QString m_userName;
};
